addons/website/static/src/common/website_visitor_model.js

```diff
@@ -0,0 +1,16 @@
+import { fields, Record } from "@mail/core/common/record";
+
+export class WebsiteVisitor extends Record {
+    static _name = "website.visitor";
+    static id = "id";
+
+    partner_id = fields.One("Persona");
+    country = fields.One("res.country", {
+        /** @this {import("models").WebsiteVisitor} */
+        compute() {
+            return this.partner_id?.country ?? this.country_id;
+        },
+    });
```

Comment:
Should define `country_id` as a `fields.One("res.country")`, otherwise the `this.country_id` would store plain data which is definitely not what we want.